lib/Extension/CodeTransformExtra/Rpc/GenerateAccessorHandler.php

```diff
@@ -42,18 +58,72 @@ public function configure(Resolver $resolver)
 
     public function handle(array $arguments)
     {
-        $sourceCode = $this->generateAccessor->generateAccessor(
-            SourceCode::fromStringAndPath(
-                $arguments[self::PARAM_SOURCE],
-                $arguments[self::PARAM_PATH]
-            ),
-            $arguments[self::PARAM_OFFSET]
-        );
+        $class = $this->class($arguments[self::PARAM_SOURCE]);
+
+        $this->requireInput(ListInput::fromNameLabelChoices(
+            self::PARAM_NAMES,
+            sprintf('Properties from "%s"', $class->name()),
+            $this->propertiesChoices($class)
+        )->withMultiple(true));
+
+        if ($this->hasMissingArguments($arguments)) {
+            return $this->createInputCallback($arguments);
+        }
+
+        $originalSource = $arguments[self::PARAM_SOURCE];
+        $newSource = SourceCode::fromStringAndPath($originalSource, $arguments[self::PARAM_PATH]);
+
+        foreach ((array) $arguments[self::PARAM_NAMES] as $propertyName) {
+            $newSource = $this->generateAccessor->generate(
+                $newSource,
+                $propertyName,
+                $arguments[self::PARAM_OFFSET]
+            );
+        }
 
         return UpdateFileSourceResponse::fromPathOldAndNewSource(
-            $sourceCode->path(),
-            $arguments[self::PARAM_SOURCE],
-            (string) $sourceCode
+            $newSource->path(),
+            $originalSource,
+            (string) $newSource
         );
     }
+
+    /**
+     * @param string $source
+     *
+     * @return ReflectionClass
+     */
+    private function class($source)
+    {
```

Comment:
This method is duplicated from `OverrideMethodHandler`. I'm thinking it would be better to not repeat this.

Would it be worth creating some, "introspection" helper/s (e.g. `ClassAtOffset`) and implement it in both this and the override method handler? Or maybe there is a better way? Could simply be something local to this extension to start with.

Reply:
What about adding it in the reflector interface ?
We already have some variation like `reflectClassLike`, what about a `reflectClassAt` ?

Reply:
The current way would be to use the collection for that (e.g. `$collection->atPosition(123)` -- plausibly it could be done for all collections, but unsure of the implementation. The general idea is to have all the "query" methods on the collection.

Reply:
ah, found an existing example :) https://github.com/phpactor/worse-reflection/blob/master/lib/Bridge/TolerantParser/Reflection/Collection/ReflectionMemberCollection.php#L39

Reply:
Ok, two things that came to mind.
Would not it be better to also add `Position::contains()` method, it feels like it's not the first time I see this condition in different places.

Does the class reflector handle anonymous class as well ?
I was thinking about this case this morning:
```php
class Foo
{
    public function foo()
    {
        $bar = new class {
            // <> cursor here
        };
    }
}
```